core/lib/types/src/base_token_ratio.rs

```diff
@@ -13,10 +15,39 @@ pub struct BaseTokenRatio {
 }
 
 /// Struct to represent API response containing denominator, numerator, and timestamp.
-#[derive(Debug)]
-pub struct BaseTokenAPIRatio {
-    pub numerator: NonZeroU64,
-    pub denominator: NonZeroU64,
-    /// Either the timestamp of the quote or the timestamp of the request.
+#[derive(Debug, Clone, PartialEq)]
+pub struct BaseTokenAPIPrice {
+    pub base_token_price: BigDecimal,
+    pub eth_price: BigDecimal,
```

Comment:
...with a single request you will also be able to use `Fraction` right there, on the API client side, and keep `numerator`/`denominator` structure you had before.